File: op/2semester/lab1/cpp/module.cpp
#include <iostream>
#include <fstream>
#include <string>
#include "module.h"

using namespace std;

string getFileName() {
    string _fileName;

    cout << "Enter file name: ";
    cin >> _fileName;

    return _fileName;
}

string getInput() {
    string _content, line;

    cout << "Write your text ( to end enter Ctrl + X ): " << endl;
    cin.ignore();
    getline(cin, line);

    while (int(line[0]) != 24) {  //
        _content += line + '\n';
        getline(cin, line);
    }

    return _content;
}

writingMode getWritingMode() {
    writingMode _wm = writeFile;
    string _input;

    cout << "Please chose a writing mode ( W - rewrite/write | E - extend) : ";
    cin >> _input;

    if (_input == "W") {

    }
    else if (_input == "E") {
        _wm = extendFile;
    }
    else {
        cout << "Incorrent mode, writing mode set to default (Write file)";
    }

    return _wm;
};

void writeToFile(string _fileName, string _content, writingMode _wm) {
    ofstream file(_fileName, _wm == writingMode::writeFile ? ios::trunc | ios::out : ios::app );
    file << _content;

    file.close();

}

void exportData(string _fileName) {
    string line;
    ifstream file(_fileName, ios::in);
    ofstream fileExp(_fileName.substr(0, _fileName.length() - 4) + "_exp.txt", ios::trunc | ios::out);

    while (!file.eof()) {
        getline(file, line);

        if (line == "") continue;

        int counter = 0;

        while (line[counter] != ' ' && counter < line.length())
        {
            counter++;
        }

        line += line[counter - 1];

        fileExp << line << endl;
    };

    file.close();
    fileExp.close();
}


void showFileContent(string _fileName, string _message) {
    string line;
    ifstream file(_fileName, ios::in);
    cout << endl << _message << endl;

    while (!file.eof()) {
        getline(file, line);
        cout << line <<endl; 
    }
    file.close();

}

File: op/2semester/lab1/python/module.py
def get_input():
    print('Write your text ( to end enter Ctrl + X ): ')
    content = ''
    line = input()
    while line != '\x18':
        content += line + '\n'
        line = input()
    return content

def write_file(_file_name , _wm,  _content):
    with open(_file_name, _wm) as file:
        file.write(_content)

def get_writing_mode():
    inp = input('Please chose a writing mode ( W - rewrite/write | E - extend) : ')
    if inp == 'W':
        mode = 'w'
    elif inp == 'E':
        mode = 'a'
    else:
        mode = 'w'
        print('Incorrent mode, writing mode set to default (Write file)')
    return mode

def export_data(file_name):
    with open(file_name , 'r') as file, open(file_name[:-4]+'_exp.txt', 'w') as file_exp:
        for line in file.readlines():
            file_exp.write(line[:-1] + line.split()[0][-1] +'\n')


def show_file_content(_file_name, _message):
    print(_message)
    with open(_file_name, 'r') as file:
        print(file.read())
